spec/unit/outputs/elasticsearch/data_stream_support_spec.rb

```diff
@@ -0,0 +1,497 @@
+
+    it "does not use data-streams on LS 7.x" do
+      change_constant :LOGSTASH_VERSION, '7.10.0' do
+        expect( subject.data_stream_config? ).to be false
+      end
+    end
+
+    it "defaults to using data-streams on LS 8.0" do
+      change_constant :LOGSTASH_VERSION, '8.0.0' do
+        expect( subject.data_stream_config? ).to be true
+      end
+    end
+
+    context 'non-compatible ES' do
+
+      let(:es_version) { '7.8.1' }
+
+      it "raises when running on LS 8.0" do
+        change_constant :LOGSTASH_VERSION, '8.0.0' do
+          error_message = /data_stream is only supported since Elasticsearch 7.9.0 \(detected version 7.8.1\)/
+          expect { subject.data_stream_config? }.to raise_error(LogStash::ConfigurationError, error_message)
+        end
+      end
+
+    end
+
+  end
+
+  context "ds-compatible configuration" do
+
+    let(:options) do
+      {
+          'hosts' => [ 'http://127.0.0.1:12345' ],
+          'http_compression' => 'true', 'bulk_path' => '_bulk', 'timeout' => '30',
+          'user' => 'elastic', 'password' => 'ForSearch!', 'ssl' => 'false'
+      }
+    end
+
+    before { allow(subject).to receive(:last_es_version).and_return(es_version) }
+
+    it "does not use data-streams on LS 7.x" do
+      change_constant :LOGSTASH_VERSION, '7.10.0' do
+        expect( subject.data_stream_config? ).to be false
+      end
+    end
+
+    it "defaults to using data-streams on LS 8.0" do
+      change_constant :LOGSTASH_VERSION, '8.0.1' do
+        expect( subject.data_stream_config? ).to be true
+      end
+      change_constant :LOGSTASH_VERSION, '8.1.0' do
+        expect( subject.send(:check_data_stream_config!) ).to be true
+      end
+    end
+
+    # it "warns about not using data-streams on LS 8.0 (OSS)" do
+    #   expect( subject.logger ).to receive(:warn) do |msg|
+    #     msg.index "Configuration is data_stream compliant but won't be used"
+    #   end
+    #   change_constant :LOGSTASH_VERSION, '8.0.1' do
+    #     change_constant :OSS, true, target: LogStash do
+    #       expect( subject.data_stream_config? ).to be false
+    #     end
+    #   end
+    # end
+    it "uses data-streams on LS 8.0 (OSS)" do
+      change_constant :LOGSTASH_VERSION, '8.0.1' do
+        change_constant :OSS, true, target: LogStash do
+          expect( subject.data_stream_config? ).to be true
+        end
+      end
+    end
+
+    context 'old ES' do
+
+      let(:es_version) { '7.8.1' }
+
+      it "fails to start on LS 8.0" do
+        change_constant :LOGSTASH_VERSION, '8.0.0' do
+          change_constant :OSS, false, target: LogStash do
+            expect { subject.data_stream_config? }.to raise_error(LogStash::ConfigurationError, /data_stream is only supported since Elasticsearch/)
+          end
+        end
+      end
+
+      # it "uses index-ing on OSS LS 8.0" do
+      #   change_constant :LOGSTASH_VERSION, '8.0.0' do
```

Comment:
:scissors: here as a reminder that we diverged from the original plan

> Logstash 8.0 OSS + Data Streams compatible settings: Default to index, warn / log to suggest use of default ES + data stream for better experience

we now behave the same regardless of whether LS is OSS